### ISSUE TITLE
Blob trigger with large file results in timeout

### ISSUE THREAD

**Issue by JonathanGiles** [Azure/azure-functions-java-library#42]
I have a BlobTrigger Java function which I believe is timing out when deployed to Azure after 5 minutes because it takes too long for the function to complete. I say this because I see the first log method `Executing 'Functions.JavaDocBlobWatcher' (Reason='New blob detected: incoming/output.zip', Id=0f54292e-460e-4c2b-8e10-f9489fe2f34b)`, but then I see a second message `Timeout value of 00:05:00 exceeded by function 'Functions.JavaDocBlobWatcher' (Id: '0f54292e-460e-4c2b-8e10-f9489fe2f34b'). Initiating cancellation.`

The file is 38.1 MB and I believe it is all loaded into the byte array that the BlobTrigger is associated with. From a brief bit of testing I believe that large blobs of data take a very long time to be sent into the function, and I wonder whether something other than a byte[] should be used (e.g. some kind of input stream).

**Comment by brunoborges** [comment]
Not only due to the possible timeout, an InputStream should be used anyways. There is no way to adjust the memory of the JVM used behind the scenes on Azure Functions, and therefore OutOfMemory could easily happen for large files.

**Comment by brunoborges** [comment]
Might be related: #23

**Comment by brunoborges** [comment]
I think this would require significant change and delay GA a lot.

My recommendation is to add a file limit on the trigger side, or a warning, and figure this out in another release.

**Comment by JonathanGiles** [comment]
The issue with delaying is that it will then be a breaking API change. That's fine if it is understood and accepted.

As an aside, my use case is such that I want to be **notified** of a new file....but I don't actually need to be **sent** the byte array of its content (I can grab that myself). I wish there was a way to opt in to being sent the content (again, a stream would help here).

**Comment by pragnagopa** [comment]
We only allow 32MB on consumption plan [Host/ScriptHost.cs#L76](https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/Host/ScriptHost.cs#L76).
@JonathanGiles - can you try on a dedicated app?

**Comment by pragnagopa** [comment]
We are tracking adding stream support for out of process languages https://github.com/Azure/azure-functions-host/issues/1361 but not for GA.

**Comment by JonathanGiles** [comment]
That's an interesting insight, thanks for sharing Pragna. If that is to be hardcoded then we should definitely improve the messaging in the logs to clarify why the function has timed out (it's more precisely not even tried), and I would wonder if we could have an API that allows for the user to opt-out of being sent the content. I can imagine one approach is to have an extra attribute on `@BlobTrigger` along the lines of `sendContent` which is by default true, and can be set to false, in which case the content array will be empty, but it will handle any file size.

**Comment by JonathanGiles** [comment]
Thanks for the stream support link - I'll keep an eye on it as it seems like a big gap we should offer across all functions SDKs.

**Comment by pragnagopa** [comment]
Yes. Correct. I will leave this issue open for improving the error message. This value is configurable for dedicated apps. Here is the issue for not reading the data from the blog https://github.com/Azure/azure-functions-host/issues/2698

**Comment by JonathanGiles** [comment]
Thanks - subscribed to that issue too!

**Comment by asavaritayal** [comment]
Related to https://github.com/Azure/azure-functions-host/issues/1361

**Comment by AspenForester** [comment]
I'm seeing similar results with Azure PowerShell Functions, Blob triggers, and files as small as 4MB.  Like others have mentioned, the content of the blob is useless to me.  All I need is the trigger metadata.  After that, if I want the content of the blob, I can go get it.

**Comment by aomegax** [comment]
any news about?

**Comment by ahmedmuhsin** [comment]
I was not able to reproduce the issue. The big delay was most likely due to Host restrictions that were being imposed at the time. 



I created a function app that gets a 50MB blob, logs it, and then outputs the blob contents into another container:



```java

package com.function;

import com.microsoft.azure.functions.annotation.*;
import com.microsoft.azure.functions.*;

/**
 * Azure Functions with Azure Blob trigger.
 */
public class BlobTriggerJava1 {
    /**
     * This function will be invoked when a new or updated blob is detected at the specified path. The blob contents are provided as input to this function.
     */
    @FunctionName("BlobTriggerJava1")
    @StorageAccount("AzureWebJobsStorage")
    public void run(
        @BlobTrigger(name = "content", path = "inputcontainer/{name}", dataType = "binary") byte[] content,
        @BindingName("name") String name,
        @BlobOutput(name = "outputBlob", path = "outputcontainer/{name}_output", dataType = "binary") OutputBinding<byte[]> outputBlob,
        final ExecutionContext context
    ) {
        double sizeInMB = content.length / (1024.0 * 1024.0);
        context.getLogger().info("Java Blob trigger function processed a blob. Name: " + name + "\n  Size: " + content.length + " Bytes (" + String.format("%.2f", sizeInMB) + " MB)");
        
        // Write the same content to the output container
        outputBlob.setValue(content);
        context.getLogger().info("Blob written to output container: " + name + "_output");
    }
}
 
```




This is the test that I ran using core tools and a real storage account:


```text
[2026-01-08T17:25:10.262Z] Executing 'Functions.BlobTriggerJava1' (Reason='New blob detected(LogsAndContainerScan): inputcontainer/testfile.txt', Id=a4fce2d0-7c6d-490c-8dd5-88582580ba81)
[2026-01-08T17:25:10.264Z] Trigger Details: MessageId: b655479c-4048-4968-9118-07c500018f39, DequeueCount: 1, InsertedOn: 2026-01-08T17:25:07.000+00:00, BlobCreated: 2026-01-08T17:21:59.000+00:00, BlobLastModified: 2026-01-08T17:25:04.000+00:00
[2026-01-08T17:25:10.592Z] Java Blob trigger function processed a blob. Name: testfile.txt
  Size: 52428800 Bytes (50.00 MB)
[2026-01-08T17:25:10.594Z] Blob written to output container: testfile.txt_output
[2026-01-08T17:25:10.603Z] Function "BlobTriggerJava1" (Id: a4fce2d0-7c6d-490c-8dd5-88582580ba81) invoked by Java Worker
[2026-01-08T17:25:12.692Z] Executed 'Functions.BlobTriggerJava1' (Succeeded, Id=a4fce2d0-7c6d-490c-8dd5-88582580ba81, Duration=4843ms)
```


On Windows Consumption:
<img width="1258" height="756" alt="Image" src="https://github.com/user-attachments/assets/fda58a11-735f-458d-9d6f-cae4af2af3ca" />



On Linux Consumption:
<img width="1261" height="865" alt="Image" src="https://github.com/user-attachments/assets/5211f192-c158-4036-a7b1-e77ecd39ce45" />
